spec/classes/init_spec.rb

```diff
@@ -191,12 +190,219 @@
             'enable'     => true,
             'hasrestart' => true,
             'hasstatus'  => v[:service_hasstatus],
+            'subscribe'  => [
+              'File[/var/lib/monit]',
+              'File[monit_config_dir]',
+              'File[monit_config]',
+            ]
           })
         end
       end
     end
   end
 
+  describe 'parameter functionality' do
+    let(:facts) do
+      {
+       :osfamily                  => 'Debian',
+       :lsbdistcodename           => 'squeeze',
+      }
+    end
+
+    context 'when check_interval is set to valid integer <242>' do
+      let(:params) { { :check_interval => 242 } }
+      it { should contain_file('monit_config').with_content(/^set daemon 242$/) }
+    end
+
+    context 'when httpd is set to valid bool <true>' do
+      let(:params) { { :httpd => true } }
+      content = <<-END.gsub(/^\s+\|/, '')
```

Comment:
Very clever! How do you think it compares to using fixtures or putting the strings in a `%{}` such as in https://github.com/ghoneycutt/puppet-module-dnsclient/blob/master/spec/classes/init_spec.rb#L19-23